### PR TITLE
Closes#207 Implemented publish/unpublish for decks

### PR DESCRIPTION
**Pull request by OlehPaliuh**

### CODE REVIEW

**Review by maxturko — COMMENTED** [review]
I was not able to figure out how to publish/unpublish decks on the website, where exactly is the button located?

**Comment by OlehPaliuh** [comment]
@maxturko In the cabinet, you can see subscribed decks and if you are the owner of the deck, there will be "PUBLIC" or "PRIVATE" button after "Start" button. Also, you can visit "Manege decks", after that chose some deck and on the right side of the screen you will see "PRIVATE" or "PUBLIC" button.

File: src/app/page/cabinet/cabinet.component.html [hunk 1]
```diff
@@ -53,10 +53,18 @@
...
         </td>
+        <td *ngIf="isOwnerDeck(deck)"></td>
+        <td *ngIf="deck.hidden===true && isOwnerDeck(deck)">
+          <button type=" button" class="btn btn-danger btn-sm" (click)="changeAccessDeck(deck, false)">PRIVATE</button>
+        </td>
+        <td *ngIf="deck.hidden===false && isOwnerDeck(deck)">
+          <button type=" button" class="btn btn-primary btn-sm" (click)="changeAccessDeck(deck, true)">PUBLIC</button>
+        </td>
+        <td *ngIf="!isOwnerDeck(deck)"></td>
+        <td *ngIf="!isOwnerDeck(deck)"></td>
```
Comment (antonbakalets) [inline src/app/page/cabinet/cabinet.component.html]: Twice the same expression. is it a typo?

File: src/app/page/cabinet/cabinet.component.ts [hunk 2]
```diff
@@ -113,17 +116,28 @@ export class CabinetComponent implements OnInit {
...
 
     this.cabinetService.updateCourse(course)
       .subscribe((response) => console.log());
   }
 
+  changeAccessDeck(deck: DeckLinkByCategory, access: boolean) {
+     deck.hidden = access;
+
+    this.cabinetService.toggleDeck(deck)
+      .subscribe((response) => console.log());
```
Comment (antonbakalets) [inline src/app/page/cabinet/cabinet.component.ts:138]: Use the fact
`import {NGXLogger} from 'ngx-logger';`
istead of 
`console.log()`

File: src/app/page/cabinet/cabinet.service.ts [hunk 3]
```diff
@@ -48,6 +48,11 @@ export class CabinetService {
       .map((response: Response) => this.logger.log(response));
   }
 
+  public toggleDeck(deck: DeckLinkByCategory) {
+    return this.orlp.put('/api/cabinet/decks/' + deck.deckId + '/toggle/access', {})
+      .map((response: Response) => this.logger.log(response));
```
Comment (antonbakalets) [inline src/app/page/cabinet/cabinet.service.ts:53]: Here the logger is used correctly.

File: src/app/page/cabinet/userDecks/user.decks.component.html [hunk 4]
```diff
@@ -58,13 +58,20 @@
...
               class="btn btn-primary decks-list-button" data-toggle="modal" data-target="#create-edit-deck">Manage Cards
       </button>
-      <button class="btn btn-primary decks-list-button" (click)="downloadCards(selectedDeck.deckId, selectedDeck.name)">Download cards
+      <button class="btn btn-primary decks-list-button" (click)="downloadCards(selectedDeck.deckId, selectedDeck.name)">
+        Download cards
       </button>
       <button class="btn btn-primary decks-list-button" (click)="downloadCardsTemplate()">Download cards template
       </button>
+      <button *ngIf="selectedDeck.hidden===true && isOwnerDeck(selectedDeck)" class="btn btn-danger decks-list-button"
+              (click)="changeAccessDeck(selectedDeck, false)">PRIVATE</button>
```
Comment (antonbakalets) [inline src/app/page/cabinet/userDecks/user.decks.component.html]: i'm not sure if UPPERCASE is acceptable in user messages.

File: src/app/page/cabinet/cabinet.component.html (hunk 1 above)
Comment (OlehPaliuh) [reply src/app/page/cabinet/cabinet.component.html]: Thanks, I corrected this mistake

File: src/app/page/cabinet/userDecks/user.decks.component.html (hunk 4 above)
Comment (OlehPaliuh) [reply src/app/page/cabinet/userDecks/user.decks.component.html]: In this case, I agree with you, so I corrected to lowercase.

File: src/app/page/cabinet/cabinet.component.ts (hunk 2 above)
Comment (OlehPaliuh) [reply src/app/page/cabinet/cabinet.component.ts:138]: Tanks, I changed that.

**Review by antonbakalets — APPROVED** [review]

File: src/app/page/cabinet/userDecks/user.decks.component.ts [hunk 5]
```diff
@@ -33,7 +35,10 @@ export class UserDecksComponent implements OnInit {
   public allSynthaxes: String[] = ['' , 'JAVA', 'HTML', 'C++', 'C#', 'JavaScript', 'TypeScript',
     'PHP', 'Go', 'Swift', 'Scala', 'Pascal', 'Ruby', 'SQL', 'C', 'ObjectiveC'];
 
-  constructor(private userDecksService: UserDecksService, private orlp: ORLPService, private deckService: DeckService) {
+  constructor(private userDecksService: UserDecksService,
+              private cabinetService: CabinetService,
+              private orlp: ORLPService,
+              private deckService: DeckService) {
```
Comment (maxturko) [inline src/app/page/cabinet/userDecks/user.decks.component.ts]: Your version is not building successfully

The errors I get are:
ERROR in D:/Project/orlp-web/src/app/page/cabinet/userDecks/user.decks.component.ts (118,37): Property 'logger' does not exist on type 'UserDecksComponent'.
ERROR in D:/Project/orlp-web/src/app/page/cabinet/userDecks/user.decks.component.ts (118,37): Property 'logger' does not exist on type 'UserDecksComponent'.

webpack: Failed to compile.

I suggest adding "private logger: NGXLogger" to this constructor and replacing line 12 with: "import {NGXLogger} from 'ngx-logger';"

**Review by maxturko — CHANGES_REQUESTED** [review]

File: src/app/page/cabinet/userDecks/user.decks.component.ts (hunk 5 above)
Comment (OlehPaliuh) [reply src/app/page/cabinet/userDecks/user.decks.component.ts]: Thanks, I corrected that.

**Review by Maks-Chornyi — APPROVED** [review]

**Review by OleksandrBartosh — APPROVED** [review]

**Review by maxturko — APPROVED** [review]

**Pull request merged**